kotlin_spark/src/main/java/com/sleazyweasel/eboshi/Auth.kt

```diff
@@ -0,0 +1,68 @@
+package com.sleazyweasel.eboshi
+
+import org.jose4j.jwk.RsaJsonWebKey
+import org.jose4j.jwk.RsaJwkGenerator
+import org.jose4j.jws.AlgorithmIdentifiers
+import org.jose4j.jws.JsonWebSignature
+import org.jose4j.jwt.JwtClaims
+import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import org.mindrot.jbcrypt.BCrypt
+
+private val TOKEN_ISSUER = "Kotlin Eboshi API"
+private val TOKEN_AUDIENCE = "eboshi user"
+private val EMAIL_CLAIM_KEY = "email"
+
+open class Auth {
+    private val rsaWebKey: RsaJsonWebKey;
+
+    constructor() {
+        rsaWebKey = RsaJwkGenerator.generateJwk(2048)!!
```

Comment:
This is an interesting idea... generate a new random key every time. I guess the downside is that everyone's sessions will be invalid after a server restart. Very secure, though, and probably good enough for a toy app.